src/android/java/com/pspdfkit/cordova/action/annotation/RemoveAnnotationAction.java

```diff
@@ -0,0 +1,73 @@
+package com.pspdfkit.cordova.action.annotation;
+
+import androidx.annotation.NonNull;
+
+import com.pspdfkit.annotations.AnnotationProvider;
+import com.pspdfkit.cordova.CordovaPdfActivity;
+import com.pspdfkit.cordova.PSPDFKitCordovaPlugin;
+import com.pspdfkit.cordova.action.BasicAction;
+import com.pspdfkit.document.PdfDocument;
+import com.pspdfkit.ui.PdfFragment;
+
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.PluginResult;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+import static com.pspdfkit.cordova.Utilities.getAnnotationTypeFromString;
+
+/**
+ * Removes a given annotation from the current document. The annotation should be in the Instant
+ * JSON format: https://pspdfkit.com/guides/ios/current/importing-exporting/instant-json/#instant-annotation-json-api
+ */
+public class RemoveAnnotationAction extends BasicAction {
+
+  private static final int ARG_ANNOTATION_JSON = 0;
+
+  public RemoveAnnotationAction(@NonNull String name, @NonNull PSPDFKitCordovaPlugin plugin) {
+    super(name, plugin);
+  }
+
+  @Override
+  protected void execAction(JSONArray args, CallbackContext callbackContext) throws JSONException {
+    JSONObject jsonObject = args.getJSONObject(ARG_ANNOTATION_JSON);
+    // We can't create an annotation from the Instant JSON since that will attach it to the document,
+    // so we manually grab the necessary values.
+    int pageIndex = jsonObject.getInt("pageIndex");
+    String name = jsonObject.getString("name");
+    String type = jsonObject.getString("type");
+
+    CordovaPdfActivity pdfActivity = CordovaPdfActivity.getCurrentActivity();
+
+    // Capture the given callback and make sure it is retained in JavaScript too.
+    final PluginResult result = new PluginResult(PluginResult.Status.NO_RESULT);
+    result.setKeepCallback(true);
+    callbackContext.sendPluginResult(result);
+
+    final PdfDocument document = pdfActivity.getDocument();
+    if (document != null) {
+      AnnotationProvider annotationProvider = document.getAnnotationProvider();
+      annotationProvider.getAllAnnotationsOfType(getAnnotationTypeFromString(type), pageIndex, 1)
+          .filter(annotationToFilter -> name.equals(annotationToFilter.getName()))
```

Comment:
I assume the name holds the UUID of the annotation? Our JSON specification describes the `name` property to be optional. Would it make sense to add another check for `pdfObjectId` (which might be `null` too however) to increase detection rate?

Reply:
@davidschreiber can't find the exact name or mention for the ID. Is it `"id": "01CHG7QMTDT1JFYQ8BSKGT6F3P"` from this guide page https://pspdfkit.com/guides/android/current/importing-exporting/instant-json/?

Reply:
Here's the spec: https://pspdfkit.com/guides/server/current/server-api/json-format/#base-annotation-type

It should be the `pdfObjectId` property. Yeah, it should map the PDF object number you can retrieve using `Annotation#getObjectNumber()`

Reply:
@davidschreiber I checked on `pdfObjectId` and it seems that it isn't mapped into `getObjectNumber` and is used for something else internally. `getObjectNumber ` returns its own different value after `createAnnotationFromInstantJson` gets called.

Reply:
I added the check for optional name condition and went one more time over react native logic for the same case and in here only the name property used as well. Is it fine if we will keep it AS-IS for now?